### ISSUE TITLE
Download manager fails to preserve login context

### ISSUE THREAD

**Issue by badger200** [opa334/SafariPlus#31]
1. It seems anything I download that I had to log in to view, download manager is “dumb” and **always acts like I’m logged out/unauthorized.** I would’ve thought my login session/cookies would be valid across Safari tabs into Download Manager. 

Is Download Manager only valid for public downloads?

2. A related but separate issue: downloading iOS mobileprovisioning profiles (.mobileconfig) fail for this reason unless download manager is disabled. Bc iOS seems to ONLY allow .mobileconfig profiles to be opened in Safari and with download manager, even tapping “Open” is not equivalent. 

My ISP Spectrum has a nice USA nationwide free broadband WiFi at http://wifiprofile.spectrum.net (login required), it’s incredible how many remote places have popped up with the 6 CableWiFi / SpectrumPlus SSIDs included in this profile!! I have no idea how they do it! Even some tiny beach coffee shops and doctors office buildings have it.

**Comment by opa334** [comment]
You're right, this unfortunately does not seem to be fixable with the APIs that apple provides. I have spent a ton of time trying to fix this issue already and even got some C++ methods to get the credentials from WebKit to not crash, but they were just empty for some reason. I'm pretty sure this is due to the web content being in a separate process, I could be wrong tho. The only way I currently see to fix this on iOS 13 and above would be to reconstruct the download manager in a way to use the stock one under the hood.

I also noticed the second issue and it might be fixable by figuring out what makes iOS open the mobileconfig dialog and make the "Open" option support it, I will look into it soon.
